include/cuco/operator.hpp

```diff
@@ -17,65 +17,360 @@
 #pragma once
 
 namespace cuco {
+// TODO point to container<->operator support matrix
+/**
+ * @brief Namespace containing device API operator tags
+ *
+ * This namespace defines tag types that are used to specify different operations
+ * that can be performed on container reference objects in device code. These
+ * tags serve as compile-time indicators for the type of operation being requested
+ * in device code, and are not intended for use with host-side bulk APIs.
+ *
+ * Example Usage:
+ * When applied to container reference object, e.g., `container_ref.rebind_operators(cuco::insert)`,
+ * it enables using `container_ref.insert(...)`.
+ * For a full example on how to use operators see `examples/static_set/device_ref_example.cu`.
+ *
+ * Available operators:
+ * - `insert`: Inserts an element into the container
+ * - `insert_and_find`: Inserts an element and returns an iterator to the stored location
+ * - `insert_or_assign`: Inserts a new element or updates an existing element
+ * - `insert_or_apply`: Inserts a new element or applies a user-defined function to an existing
+ * element
+ * - `erase`: Removes an element from the container
+ * - `contains`: Checks element existence
+ * - `count`: For a given key, returns the number of matching elements in the container
+ * - `find`: Locates an element in the container
+ * - `retrieve`: Retrieves all matching elements in the container for a given key
+ * - `for_each`: Applies a user-defined function to each element in the container (or for a given
+ * key)
+ *
+ */
 inline namespace op {
-// TODO enum class of int32_t instead of struct
-// https://github.com/NVIDIA/cuCollections/issues/239
+
 /**
- * @brief `insert` operator tag
+ * @brief Tag type for `insert` operator
+ *
+ * API Signature:
+ * ```cpp
+ * template <typename Value>
+ * __device__ bool insert(Value const& value) noexcept
+ *
+ * template <typename Value>
+ * __device__ bool insert(cooperative_groups::thread_block_tile<cg_size> const& group,
+ *                        Value const& value) noexcept
+ * ```
```

Comment:
Based on the offline discussion. An alternative to avoid `@see` is to move parameter docs into the doc cpp code samples.